System/Posix/Files/Common.hsc

```diff
@@ -294,6 +305,19 @@ modificationTime (FileStatus stat) =
 statusChangeTime (FileStatus stat) =
   unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_ctime)
 
+#ifdef HAVE_STRUCT_STAT_ST_BLOCKS
+fileBlocks (FileStatus stat) =
+  Just $ unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_blocks)
+#else
+fileBlocks _ = Nothing
+#endif
+#ifdef HAVE_STRUCT_STAT_ST_BLKSIZE
+fileBlockSize (FileStatus stat) =
+  Just $ unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_blksize)
```

Comment:
Same idea as above, and again, I'm not actually sure which is more corrrect. @Bodigrim your take?
```suggestion
  unsafePerformIO $ fmap Just $ withForeignPtr stat $ (#peek struct stat, st_blksize)
```

Reply:
The above is likely (intended to be) equivalent to `Just $! unsafePerformIO $ ...`